File: JAVA Programming/InsertionSort.java
public class InsertionSort {
    
    public static void Insertion_Sort(int A[],int n)
    {
        int i,j,k;

        for(i=1;i<n;i++)
        {
            k = A[i];  j = i-1;

            while(j>=0 && k<A[j])
            {
                A[j+1] = A[j];
                j--;
            }

            A[j+1] = k;
        }
    }

    public static void main(String[] args) {
        int arr[] = {4,7,9,21,3,4,5,1};    // size 8 ;
        int n = 8;
         System.out.println("Print Array Before Sorting ");
         for (int e : arr) {
 
             System.out.print("\t"+e);
             
         }
 
         System.out.println("\nPrint Array After Sorting ");
         Insertion_Sort(arr, n);
         for (int e : arr) {
 
             System.out.print("\t"+e);
             
         }
    }
}

File: C++ Programming/Insertion_sort.cpp
#include <iostream>
using namespace std;
void Print(int *arr, int size)
{
    int i = 0;
    cout << "The Array is --- \n";
    for (i = 0; i < size; i++)
        cout << arr[i] << "\t";
    cout << endl<< endl;
}
void Insertionsort(int* arr,int size)
{
    int i,j,k,temp;
    for(i=1;i<size;i++)
    {
        j = i-1; k = arr[i];
        while(j >= 0 && k < arr[j])
        {
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = k;
    }
}
int main(int argc, char const *argv[])
{
     int size = 10;
    int arr[size] = {1,56,23,12,5,67,321,567,53,20,37};
    Print(arr,size);
    cout<<"After the Selection sort \n";
    Insertionsort(arr,size);
    Print(arr,size);
    return 0;
}
